src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/OrientationBiasFilterer.java

```diff
@@ -0,0 +1,361 @@
+        final Map<VariantContext, List<Genotype>> newGenotypes = new HashMap<>();
+
+        // For each sample, perform the actual filtering.
+        for (final String sampleName: sampleNames) {
+
+            // Count the total number of unfiltered genotypes for this sample, regardless of artifact mode or not.
+            //  I.e. only remove filtered or ref/ref genotypes and count the rest.
+            final long unfilteredGenotypeCount = OrientationBiasUtils.calculateUnfilteredNonRefGenotypeCount(preAdapterQAnnotatedVariants, sampleName);
+
+            final SortedMap<Genotype, VariantContext> genotypesToConsiderForFiltering = sampleNameToVariants.get(sampleName);
+
+            // Save some time, especially for the normal sample
+            if (genotypesToConsiderForFiltering.keySet().size() == 0) {
+                logger.info(sampleName + ": Nothing to filter.");
+                continue;
+            }
+
+            // Populate counts of artifact modes.  These are genotypes that we can potentially cut.
+            final Map<ArtifactMode, Long> artifactModeCount = createArtifactModeCountMap(relevantArtifactModes, genotypesToConsiderForFiltering);
+
+            // Global number of artifacts to cut (not adjusted for preAdapterQ)
+            final Map<ArtifactMode, Long> artifactModeNumToCut = createArtifactModeToNumCutPrePreAdapterQ(fdrThresh, sampleName, unfilteredGenotypeCount, genotypesToConsiderForFiltering, artifactModeCount);
+
+            // Adjust the artifact mode to cut based on the preAdapterQ score from picard
+            for (final ArtifactMode artifactMode : artifactModeNumToCut.keySet()) {
+                double suppression;
+                if (relevantArtifactModesWithoutComplements.contains(artifactMode)) {
+                    suppression = ArtifactStatisticsScorer.calculateSuppressionFactorFromPreAdapterQ(preAdapterQScoreMap.get(artifactMode));
+                } else {
+
+                    // Must be a complement, since it cannot be in relevantArtifactModes while not being in relevantArtifactModesWithoutComplements
+                    suppression = ArtifactStatisticsScorer.calculateSuppressionFactorFromPreAdapterQ(preAdapterQScoreMap.get(OrientationBiasUtils.createReverseComplement(artifactMode)));
+                }
+
+                artifactModeNumToCut.put(artifactMode,
+                        Math.round(artifactModeNumToCut.get(artifactMode) * suppression));
+                logger.info(sampleName + ": Cutting (" + artifactMode.toString() +
+                        ") post-preAdapterQ: " + String.valueOf(artifactModeNumToCut.get(artifactMode)));
+            }
+
+            // Add filtering results to genotypes and store a pair of genotype variant context, so that we can update variant contexts later.
+            logger.info(sampleName + ": Adding orientation bias filter results to genotypes...");
+
+            final Map<ArtifactMode, Long> artifactModeCutSoFar = new HashMap<>();
```

Comment:
Not sure I see the advantage.   Especially, since I get the value a lot and I will be constantly casting as Integer.

No action, but if you feel strongly....